### PR TITLE
stb167  Refactor config

### PR DESCRIPTION
**Pull request by duker33**
https://github.com/fidals/stroyprombeton/issues/167

### CODE REVIEW

File: docker/create-env.sh [hunk 1]
```diff
@@ -24,23 +24,29 @@ function join_by { local IFS="$1"; shift; echo "$*"; }
...
         then
             cp $file $new_file
         fi
+    }
+
+    for file in env_files/*.dist
+    do
+        new_file $file
     done
+    new_file ../shopelectro/settings/local.py.dist
```
Comment (artemiy312) [inline docker/create-env.sh]: The `create_env_files` function in `create-env.sh` may not create any configs. The single responsibility of this script is create env files, so let's do it elsewhere

Comment (artemiy312) [reply docker/create-env.sh]: Also `.env` file will be contain the content of `../shopelectro/settings/local.py.dist` (see a 71 line) and it is absolutely unexpected behavior

File: shopelectro/settings/prod.py [hunk 2]
```diff
@@ -1,74 +0,0 @@
...
-REDIS_PASSWORD = os.environ['REDIS_PASSWORD']
-
-REDIS_DSN = f'redis://{REDIS_URL}:{REDIS_PORT}/'
-
-REDIS_LOCATION_DEFAULT = os.environ['REDIS_LOCATION_DEFAULT']
-REDIS_LOCATION_SESSION = os.environ['REDIS_LOCATION_SESSION']
-REDIS_LOCATION_THUMBNAIL = os.environ['REDIS_LOCATION_THUMBNAIL']
-REDIS_LOCATION_USER_AGENT = os.environ['REDIS_LOCATION_USER_AGENT']
-
-CACHES = {
```
Comment (artemiy312) [inline shopelectro/settings/prod.py]: I did not understand the point. Where will we store these configs?

File: docker/create-env.sh [hunk 3]
```diff
@@ -51,16 +57,22 @@ Are you ready to continue and build a new .env? [y/n]: " yn
         fi
     fi
 
-    local env_doc=\
+    if [[ ! -f ".env" ]]
```
[the remark this answers is not in the document]
Comment (duker33) [reply docker/create-env.sh]: hmm, it's true we should always overrite it, sorry

File: docker/create-env.sh [hunk 4]
```diff
@@ -51,16 +57,22 @@ Are you ready to continue and build a new .env? [y/n]: " yn
...
+    then
+        echo "generate .env file"
```
[the remark this answers is not in the document]
Comment (duker33) [reply docker/create-env.sh]: we'll remove if, so, we'll remove all nested echo's too

TODO

File: docker/create-env.sh (hunk 1 above)
Comment (duker33) [reply docker/create-env.sh]: > Also .env file will be contain

y, i'll fix it

> The create_env_files function in create-env.sh

both django settings and env files are about env customisation in terms of our app.
In terms of linux env - not, of course. But app-sided script should work with app env, not with linux env.

I can rename script to `create-local-env`. Or can double it's code to separated script, if it looks confusing

File: shopelectro/settings/prod.py (hunk 2 above)
Comment (duker33) [reply shopelectro/settings/prod.py]: somewhere out of cvs. Simple - on serv and every of us in personal vault. Complex - separated vault or separated cvs repository with configs.

We should remove it by two reasons:
- we should be able to change local (serv too) config without working with cvs
- we should be able to put some secrets to this config

File: docker/create-env.sh (hunk 1 above)
Comment (duker33) [reply docker/create-env.sh]: moved config processing to separated script

File: shopelectro/settings/prod.py (hunk 2 above)
Comment (artemiy312) [reply shopelectro/settings/prod.py]: > Simple or Complex

We should decide which one to choose. Let's create todo for this

Comment (duker33) [reply shopelectro/settings/prod.py]: created https://github.com/fidals/shopelectro/issues/332

File: docker/create-env.sh [hunk 5]
```diff
@@ -24,23 +24,28 @@ function join_by { local IFS="$1"; shift; echo "$*"; }
 
 function create_env_files {
     new_files=()
-    for file in env_files/*.dist
-    do
+    function new_file {
```
[the remark this answers is not in the document]
Comment (duker33) [reply docker/create-env.sh]: TODO

**Pull request merged**